rebase_v1.13.1_report.md

```diff
@@ -0,0 +1,38 @@
+
+## Toolchain and Dependency Shifts
+- Upstream now enforces Go **1.24+** (release notes call out 1.25.2) via `.go-version` and `GOTOOLCHAIN`. Our builders and CI images must ship that compiler, otherwise `go` will attempt to auto-download toolchains (blocked in disconnected builds).
+- `go.mod` migrates from legacy dependencies (`aws-sdk-go` v1, DataDog v1, etc.) to new major versions (`aws-sdk-go-v2`, `dd-trace-go/v2`, OpenTelemetry packages, etc.), and bumps Kubernetes libs to **v0.34.1**. These introduce breaking API changes that our OpenShift carry code (including the external plugin) must accommodate.
+- Removal of the `vendor/` directory in upstream implies all historical vendoring scripts (`dep-ensure`, Dockerfiles that copied `vendor/`) and downstream assumptions about offline builds must be rewritten.
+
+## Core Runtime and Plugin Differences
+- `plugin.cfg` in `v1.13.1` adds new built-ins (`quic`, `multisocket`, `nomad`) and exposes `COREDNS_PLUGINS` overrides. We must reinsert `ocp_dnsnameresolver` without breaking the new ordering constraints or the generated `zplugin.go` / `zdirectives.go`.
+- Core server code (`core/dnsserver/*`, `coremain/run.go`) gained QUIC/HTTPS refactors, new configuration paths, and extensive tests. OpenShift-specific patches touching these surfaces (e.g., cache behavior, bufsize handling) will likely conflict and must be forward-ported carefully.
+- The Kubernetes plugin evolved across several releases alongside controller-runtime API changes. Custom patches we carry for OpenShift bugs (OCPBUGS-15755, -27397, -34619, etc.) need a fresh review because upstream code may have diverged significantly.
+
+## Build, Release, and CI Implications
+- Upstream Dockerfile now targets `distroless/static-debian12`, adds `--no-install-recommends`, and resets `WORKDIR`. Downstream `Dockerfile.ocp` and image build automation must mirror these shifts while preserving ART requirements.
+- GitHub workflows, lint configs (`.golangci.yml`), and README build guidance now assume the newer toolchain and add security automation (OpenSSF scorecard, etc.). Our disabled-dependabot carry patches may need to be re-thought or reapplied.
+- `make check` now relies on `go generate` followed by `go get` under modules. In disconnected builds this may fail unless we provide a vendored module proxy or prepopulated module cache.
+
+## Risk Hotspots & Open Questions
+- Compatibility of `github.com/openshift/coredns-ocp-dnsnameresolver` with the upgraded Kubernetes client libraries and Go runtime is unknown; it may require upstream updates before the rebase can land.
+- Dropping the vendor tree while keeping offline build guarantees for OpenShift release tooling will require alternative dependency management (e.g., `go mod vendor` post-rebase or mirrored module proxies).
```

Comment:
Rerunning `go mod vendor` is a normal part of our process.  That is, we always re-do the `UPSTREAM: <carry>: openshift: Vendor required modules` commit.

Reply:
👍